client/client.go

```diff
@@ -0,0 +1,64 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"istio.io/manager/model"
+)
+
+type ManagerClient struct {
+	base             *url.URL
+	versionedAPIPath string
+	client           *http.Client
+}
+
+func NewManagerClient(base *url.URL, apiVersion string, client *http.Client) *ManagerClient {
+	trimmedBase := base
```

Comment:
Its ensuring we are consistent with what we use as the url and apiversion regardless of whether people pass us stuff with trailing and leading slashes. So we don't end up making requests like: `http://url//v1/config...` or `http://url/v1//config...`.